python/ray/tune/experiment.py

```diff
@@ -49,8 +48,13 @@ def __init__(self, name, run, stop=None, config=None,
             "checkpoint_freq": checkpoint_freq,
             "max_failures": max_failures
         }
-        self._trials = generate_trials(spec, name)
 
-    def trials(self):
-        for trial in self._trials:
-            yield trial
+        self.name = name
+        self.spec = spec
+
+
+class JSONExperiment(Experiment):
```

Comment:
Can this be a function that returns an Experiment object instead, e.g.,

Experiment.from_json(name, spec)